hazelcast-jet-core/src/main/java/com/hazelcast/jet/Job.java

```diff
@@ -95,4 +95,14 @@ default void join() {
         Util.uncheckRun(() -> getFuture().get());
     }
 
+    /**
+     * Cancels the current execution if the job is currently in running and
+     * schedules a new execution with the current member list of the Jet cluster
+     *
+     * @throws IllegalStateException if the job has been already completed
+     *
+     * @return true if the current execution of the job is cancelled
```

Comment:
It would be good to mention when `false` is returned. I guess it's when job is not yet started or is already restarting.

Reply:
We already mention when it returns true: "Cancels the current execution if the job is currently in running". Do we need to mention when it returns false?

Reply:
I think it would be good. It made me wonder why wouldn't it succeed.